src/PageLayout/useHorizontalResize.tsx

```diff
@@ -0,0 +1,97 @@
+import React, {useEffect, useCallback, useState, RefObject} from 'react'
+
+export function useHorizontalResize(
+  enabled: boolean,
+  panePosition: 'start' | 'end',
+  paneRef: RefObject<HTMLDivElement>,
+  containerRef: RefObject<HTMLDivElement>,
+  storageKey?: string
+) {
+  const [isDragging, setIsDragging] = useState(false)
+  const [isResizing, setIsResizing] = useState(false)
+  const [paneWidth, setPaneWidth] = useState<undefined | number>()
+
+  useEffect(() => {
+    let hasValue = false
+    if (storageKey) {
+      const storedWidth = window.localStorage.getItem(storageKey)
```

Comment:
As @mattcosta7 pointed out in https://github.com/github/issues/issues/3652 we might want to use a wrapped around the `localStorage` call to avoid throwing an error for folks who disabled 3rd party tracking in their browser.

Reply:
Good call. Updated ✅